cmd_clean/main.go

```diff
@@ -0,0 +1,65 @@
+/* Copyright © 2022 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package main
+
+import (
+	"flag"
+	"os"
+
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/clean"
+	"github.com/vmware-tanzu/nsx-operator/pkg/config"
+	"github.com/vmware-tanzu/nsx-operator/pkg/logger"
+)
+
+// usage:
+// ./bin/clean -cluster='domain-c9:47b14f07-33b4-478b-8025-a6ae16e81be7'  -thumbprint="0E:CA:B5:98:D4:46:8C:19:4A:54:4A:67:7C:7B:9D:22:0F:78:76:5E" -log-level=0 -vc-user="administrator@vsphere.local" -vc-passwd="RAZKssvew+FyR7+7" -vc-endpoint="sc2-10-187-150-214.nimbus.eng.vmware.com" -vc-sso-domain="vsphere.local" -vc-https-port=443  -mgr-ip="10.187.157.154"
+
+var (
+	log          = logger.Log
+	cf           *config.NSXOperatorConfig
+	mgrIp        string
+	vcEndpoint   string
+	vcUser       string
+	vcPasswd     string
+	vcSsoDomain  string
+	vcHttpsPort  int
+	thumbprint   string
+	vmcaCertFile string
+	cluster      string
+)
+
+func main() {
+	flag.StringVar(&vcEndpoint, "vc-endpoint", "", "nsx manager ip")
+	flag.StringVar(&vcSsoDomain, "vc-sso-domain", "", "nsx manager ip")
+	flag.StringVar(&mgrIp, "mgr-ip", "", "nsx manager ip")
+	flag.StringVar(&vcUser, "vc-user", "", "nsx username")
+	flag.StringVar(&vcPasswd, "vc-passwd", "", "nsx password")
+	flag.IntVar(&vcHttpsPort, "vc-https-port", 443, "nsx password")
+	flag.StringVar(&thumbprint, "thumbprint", "", "nsx thumbprint")
+	flag.StringVar(&vmcaCertFile, "vmca-cert-file", "", "vmca cert file")
+	flag.StringVar(&cluster, "cluster", "", "cluster name")
+	flag.IntVar(&config.LogLevel, "log-level", 0, "Use zap-core log system.")
+	flag.Parse()
+
+	logf.SetLogger(logger.ZapLogger())
+	cf = config.NewNSXOpertorConfig()
+	cf.NsxApiManagers = []string{mgrIp}
+	cf.VCUser = vcUser
+	cf.VCPassword = vcPasswd
+	cf.VCEndPoint = vcEndpoint
+	cf.SsoDomain = vcSsoDomain
+	cf.HttpsPort = vcHttpsPort
+	cf.Thumbprint = []string{thumbprint}
+	cf.NsxApiCertFile = vmcaCertFile
+	cf.Cluster = cluster
```

Comment:
Just a Q: Is this the recommended way to build this type of config?

Reply:
Hey, do you have any other way? I'm glad to know.